### PR TITLE
drivers: mbox: cleanups and API improvements

### PR DESCRIPTION
**Pull request by gmarull**

### CODE REVIEW

**Review by carlocaione — DISMISSED** [review]
No complains

**Review by carlocaione — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Review by manuargue — DISMISSED** [review]

**Review by Dat-NguyenDuy — DISMISSED** [review]

**Review by hubertmis — DISMISSED** [review]

File: samples/drivers/mbox/boards/adp_xc7k_ae350.overlay [hunk 1]
```diff
@@ -16,4 +16,10 @@
 		compatible = "mmio-sram";
 		reg = <0x00000000 0x10000000 >;
 	};
+
+	mbox-consumer {
+		compatible = "vnd,mbox-consumer";
```
Comment (nordicjm) [inline samples/drivers/mbox/boards/adp_xc7k_ae350.overlay:21]: Needs converting to hwmv2 (check if any board names have changed and apply change)

**Comment by aescolar** [comment]
@nordicjm @gmarull if this PR needs converting to the hwmv2 please do so. In any case a rebase and push retriggering CI would be welcome.

**Comment by gmarull** [comment]
> @nordicjm @gmarull if this PR needs converting to the hwmv2 please do so. In any case a rebase and push retriggering CI would be welcome. (feel free to remove the dnm when you believe the PR is ready)

will do after https://github.com/zephyrproject-rtos/zephyr/pull/69303, since it will require another rebase otherwise.

**Review by fabiobaltieri — APPROVED** [review]

**Review by manuargue — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Pull request merged**